Document new features + docker

SamGuay opened the pull request:
Takes care of #308 and #307 (partially)

SamGuay committed:
add --do_not_reorder_entities doc

SamGuay committed:
add bids_uri doc

SamGuay committed:
fix gha + order main commands

SamGuay committed:
add readme me doc

SamGuay committed:
add do_not_reorder to advance comm

SamGuay committed:
up py3.7 to 3.8

arnaudbore commented:
- [x] Check all references to python 3.7
- [x] Section container for the commands
- [x] Put containers in installation right before install binaries

SamGuay committed:
long due update for changelog

SamGuay committed:
add container doc

SamGuay committed:
bump version

SamGuay commented:
@arnaudbore I left the docker commands to you.

I leave it to you whether to add more details re bids_validator + pydeface...

arnaudbore committed:
update container with links

arnaudbore committed:
fix typo

SamGuay commented on docs/how-to/container.md:
```diff
@@ -0,0 +1,125 @@
+
+## Step 3: Run `dcm2bids_scaffold`
+
+To run `dcm2bids_scaffold`, `with singularity`, you need to *execute* a command instead of *running* the pre-specified command (`dcm2bids`). You need to bind the respective volumes.
```
```suggestion
To run `dcm2bids_scaffold` with Apptainer/Singularity, you need to *execute* a command instead of *running* the pre-specified command (`dcm2bids`). You need to bind the respective volumes.
```

SamGuay commented on docs/how-to/container.md:
```diff
@@ -0,0 +1,125 @@
+
+## Step 4: Run `dcm2bids_helper`
+
+To run `dcm2bids_helper`, `with singularity`, you need to *execute* a command instead of *running* the pre-specified command (`dcm2bids`). To bind the respective volumes, you have two options:
```
```suggestion
To run `dcm2bids_helper` with Apptainer/Singularity, you need to *execute* a command instead of *running* the pre-specified command (`dcm2bids`). To bind the respective volumes, you have two options:
```

arnaudbore committed:
Update docs/how-to/container.md

Co-authored-by: Samuel Guay <samuel.guay@umontreal.ca>

arnaudbore committed:
Update docs/how-to/container.md

Co-authored-by: Samuel Guay <samuel.guay@umontreal.ca>

smeisler commented on docs/get-started/install.md:
```diff
@@ -9,6 +9,19 @@ date: 2022-04-17
+
+=== "Apptainer/Singularity"
+
+    `singularity pull dcm2bids_latest.sif docker://unfmontreal/dcm2bids:latest`
```
I generally think for troubleshooting purposes it is always best to specify a version when pulling containers (in case new versions are unexpectedly or unknowingly released).
```suggestion
    `apptainer pull dcm2bids_${VERSION}.sif docker://unfmontreal/dcm2bids:${VERSION}`
```
Same could be applied to the `docker pull` line above.

smeisler commented on docs/how-to/container.md:
````diff
@@ -0,0 +1,125 @@
+=== "Apptainer/Singularity"
+
+    ```
+    apptainer pull dcm2bids.sif docker://unfmontreal/dcm2bids:latest
````
(same idea as suggested above)
```suggestion
    apptainer pull dcm2bids_${VERSION}.sif docker://unfmontreal/dcm2bids:${VERSION}
```

smeisler commented on docs/how-to/container.md:
````diff
@@ -0,0 +1,125 @@
+=== "Apptainer/Singularity"
+
+    ```
+    apptainer run dcm2bids.sif --help
````
Adding safer `singularity run` arguments
```suggestion
    apptainer run -e --containall dcm2bids.sif --help
```

smeisler commented on docs/how-to/container.md:
````diff
@@ -0,0 +1,125 @@
+    --entrypoint /venv/bin/dcm2bids_scaffold \
+    -v /path/to/output-dir:/output \
+    unfmontreal/dcm2bids:latest -o /output/new_bids_dataset
+    ```
````
```suggestion
    ```
    docker run --rm -it \
    --entrypoint /venv/bin/dcm2bids_scaffold \
    -v /path/to/bids:/bids \
    unfmontreal/dcm2bids:${VERSION} -o /bids/new_scaffold
    ```
```

smeisler commented on docs/how-to/container.md:
````diff
@@ -0,0 +1,125 @@
+    singularity exec \
+    -B /path/to/output-dir:/output \
+    dcm2bids.sif dcm2bids_scaffold -o /output/new_bids_dataset
+    ```
````
```suggestion
    ```
    apptainer exec \
    -e --containall \
    -B /path/to/bids:/bids \
    dcm2bids.sif dcm2bids_scaffold -o /bids/new_scaffold
    ```
```

smeisler commented on docs/how-to/container.md:
````diff
@@ -0,0 +1,125 @@
+    -v /path/to/input-data:/input \
+    -v /path/to/output-dir/new_bids_dataset:/output \
+    unfmontreal/dcm2bids:latest -o /output -d /input
+    ```
````
```suggestion
    ```
    docker run --rm -it --entrypoint /venv/bin/dcm2bids_helper \
    -v /path/to/dicoms:/dicoms:ro \
    -v /path/to/bids/new_scaffold:/bids \
    unfmontreal/dcm2bids:${VERSION} -o /bids -d /dicoms
    ```
```

smeisler commented on docs/how-to/container.md:
````diff
@@ -0,0 +1,125 @@
+    -B /path/to/input-data:/input \
+    -B /path/to/output-dir/new_bids_dataset:/output \
+    dcm2bids.sif dcm2bids_helper -o /output -d /input
+    ```
````
```suggestion
    ```
    apptainer exec \
    -B /path/to/dicoms:/dicoms:ro \
    -B /path/to/bids/new_scaffold:/bids \
    dcm2bids.sif dcm2bids_helper -o /bids -d /dicoms
    ```
```

smeisler commented on docs/how-to/container.md:
```diff
@@ -0,0 +1,125 @@
+1. Put the input data in the same parent directory as the output directory.
+2. Specify the input data directory as a separate volume.
+
+If you bind the newly scaffolded directory on its own, you can simply use the `-o /output` instead of having to specify the full path to the scaffolded directory. Same goes for the input data directory, if the input data directory is one subject, you can bind it directly to `/input`. If it is the parent directory of multiple subjects, you can bind it to `/input` and specify the specific subject d ...
```
```suggestion
If you bind the newly scaffolded directory on its own, you can simply use the `-o /bids` instead of having to specify the full path to the scaffolded directory. Same goes for the input data directory, if the input data directory is one subject, you can bind it directly to `/dicoms`. If it is the parent directory of multiple subjects, you can bind it to `/dicoms` and specify the specific subject directory (e.g, `-d /dicoms/subject-01`).
```

smeisler commented on docs/how-to/container.md:
````diff
@@ -0,0 +1,125 @@
+    -B /path/to/output-dir/new_bids_dataset:/output \
+    unfmontreal/dcm2bids:latest --auto_extract_entities --bids_validate \
+    -o /output -d /input -c /output/code/config.json -p 001
+    ```
````
```suggestion
    ```
    docker run --rm -it \
    -v /path/to/dicoms:/dicoms:ro \
    -v /path/to/config.json:/config.json:ro \
    -v /path/to/bids/new_scaffold:/bids \
    unfmontreal/dcm2bids:${VERSION} --auto_extract_entities --bids_validate \
    -o /bids -d /dicoms -c /config.json -p 001
    ```
```

smeisler commented on docs/how-to/container.md:
````diff
@@ -0,0 +1,125 @@
+    -B /path/to/output-dir/new_bids_dataset:/output \
+    dcm2bids.sif --auto_extract_entities --bids_validate \
+    -o /output -d /input -c /output/code/config.json -p 001
+    ```
````
```suggestion
    ```
    apptainer run \
    -B /path/to/dicoms:/dicoms:ro \
    -B /path/to/config.json:/config.json:ro \
    -B /path/to/bids/new_scaffold:/bids \
    dcm2bids.sif --auto_extract_entities --bids_validate \
    -o /bids -d /dicoms -c /config.json -p 001
    ```
```

smeisler commented on docs/how-to/create-config-file.md:
```diff
@@ -149,6 +148,12 @@ WARNING:dcm2bids.structure:✅ Filename was reordered according to BIDS entity t
+
+!!! tip "`--do_not_reorder_entities`"
+
+    If you prefer to have manual control over the order of `custom_entities`, you can use the `--do_not_reorder_entities` flag. This flag allows you to keep the order defined by you, the user, in the `custom_entities` field. However, please note that this flag cannot be used in conjunction with the `--auto_extract_entities` flag.
```
```suggestion
    If you prefer to have manual control over the order of `custom_entities`, you can use the `--do_not_reorder_entities` flag. This flag allows you to keep the order defined by you, the user, in the `custom_entities` field. However, please note that this flag cannot be used in conjunction with the `--auto_extract_entities` flag. Also please keep in mind that a custom non-BIDS compliant entity order may preclude downstream usage of BIDS apps such as fMRIPrep.
```

arnaudbore committed:
Update docs/how-to/create-config-file.md

Co-authored-by: Steven Meisler <smeisler@g.harvard.edu>

SamGuay committed:
Apply suggestions from code review

Edited a bit to make sure dcm2bids_scaffold creates it in a newly created dir.

Co-authored-by: Steven Meisler <smeisler@g.harvard.edu>

SamGuay committed:
Update docs/get-started/install.md

Co-authored-by: Steven Meisler <smeisler@g.harvard.edu>

SamGuay commented on docs/get-started/install.md:
```diff
@@ -9,6 +9,19 @@ date: 2022-04-17
+
+=== "Docker"
+
+    `docker pull unfmontreal/dcm2bids:latest`
```
```suggestion
    `docker pull unfmontreal/dcm2bids:${VERSION}`
```

SamGuay commented on docs/how-to/container.md:
````diff
@@ -0,0 +1,128 @@
+=== "Docker"
+
+    ```
+    docker pull unfmontreal/dcm2bids:latest
````
```suggestion
    docker pull unfmontreal/dcm2bids:${VERSION}
```

SamGuay committed:
Apply ${VERSION} everywhere for consistency

SamGuay committed:
add missing -e --containall

SamGuay committed:
add export version to containers
